ci-operator/step-registry/ipi/install/powervs/install/ipi-install-powervs-install-commands.sh

```diff
@@ -201,6 +201,227 @@ EOF
+        ibmcloud pi service-target ${SERVICE_INSTANCE_CRN}
+	fi
+}
+function check_resources(){
+  #This function checks for any remaining DHCP leases/leftover/uncleaned resources and cleans them up before installing a new cluster
+  set +e
+  #install the tools required
+  cd /tmp
+  curl --output /tmp/IBM_CLOUD_CLI_amd64.tar.gz https://download.clis.cloud.ibm.com/ibm-cloud-cli/2.9.0/IBM_Cloud_CLI_2.9.0_amd64.tar.gz
+  tar xvzf /tmp/IBM_CLOUD_CLI_amd64.tar.gz
+  for I in infrastructure-service power-iaas cloud-internet-services cloud-object-storage dl-cli dns; do /tmp/Bluemix_CLI/bin/ibmcloud plugin install ${I}; done
+  curl -L --output /tmp/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && chmod +x /tmp/jq
+  /tmp/jq --version
+  
+  echo "Check resource phase initiated"
+  declare -a ENV_VARS
+  ENV_VARS=( "${BASE_DOMAIN}" "${dir}/" "${CLUSTER_NAME}" "${IBMCLOUD_API_KEY}" "${IBMCLOUD_APIKEY_CCM_CREDS}" "${IBMCLOUD_APIKEY_INGRESS_CREDS}" "${IBMCLOUD_APIKEY_MACHINEAPI_CREDS}" "${POWERVS_USER_ID}" "${POWERVS_REGION}" "${POWERVS_ZONE}" "${POWERVS_RESOURCE_GROUP}" "${POWERVS_SERVICE_INSTANCE_ID}" "${VPCREGION}" )
+
+    set -euo pipefail
+
+  PATH=${PATH}:$(pwd)/bin:/tmp/:/tmp/jq:/tmp/Bluemix_CLI/bin:/tmp/Bluemix_CLI/bin/ibmcloud
+  BASE64_API_KEY="$(echo -n ${IBMCLOUD_API_KEY} | base64)"
+  IC_API_KEY=${IBMCLOUD_API_KEY}
+  export PATH
+  export BASE64_API_KEY
+  export IC_API_KEY
+  init_ibmcloud
+  flag_destroy_resources=false
+
+  #Uncomment for even more debugging!
+  #export TF_LOG_PROVIDER=TRACE
+  #export TF_LOG=TRACE
+  #export TF_LOG_PATH=/tmp/tf.log
+  #export IBMCLOUD_TRACE=true
+
+  set -x
+
+  #
+  # Quota check DNS
+  #
+  if ibmcloud cis 1>/dev/null 2>&1
+  then
+    # Currently, only support on x86_64 arch :(
+    ibmcloud cis instance-set "$(ibmcloud cis instances --output json | jq -r '.[].name')"
+    DNS_DOMAIN_ID="$(ibmcloud cis domains --output json | jq -r '.[].id')"
+    export DNS_DOMAIN_ID
+    RECORDS="$(ibmcloud cis dns-records ${DNS_DOMAIN_ID} --output json | jq -r '.[] | select (.name|test("'${CLUSTER_NAME}'.*")) | "\(.name) - \(.id)"')"
+    if [ -n "${RECORDS}" ]
+    then
+      echo "${RECORDS}"
+      if [ "$flag_destroy_resources" != true ] ; then
+        flag_destroy_resources=true
+      fi
+    fi
+  fi
+
+  #
+  # Quota check cloud connections
+  #
+  CIS_INSTANCE_CRN=$(ibmcloud cis instances --output json | jq -r '.[].id');
+  export CIS_INSTANCE_CRN
+  SERVICE_INSTANCE_CRN="$(ibmcloud resource service-instances --output JSON | jq -r '.[] | select(.guid|test("'${POWERVS_SERVICE_INSTANCE_ID}'")) | .crn')"
+  export SERVICE_INSTANCE_CRN
+  ibmcloud pi service-target ${SERVICE_INSTANCE_CRN}
+  CONNECTIONS="$(ibmcloud pi connections --json | jq -r '.Payload.cloudConnections|length')"
+  if (( ${CONNECTIONS} >= 2 ))
+  then
+    echo "Error: Cannot have 2 or more cloud connections.  You currently have ${CONNECTIONS}."
```

Comment:
the code already does this. is this something extra?

Reply:
That is pre resource check since the destroy is unable to clear everything before next run